channels_redis/core.py

```diff
@@ -209,10 +208,7 @@ def __init__(
         self._receive_index_generator = itertools.cycle(range(len(self.hosts)))
         self._send_index_generator = itertools.cycle(range(len(self.hosts)))
         # Decide on a unique client prefix to use in ! sections
-        # TODO: ensure uniqueness better, e.g. Redis keys with SETNX
-        self.client_prefix = "".join(
-            random.choice(string.ascii_letters) for i in range(8)
-        )
+        self.client_prefix = uuid.uuid4().hex
```

Comment:
This is significantly more characters - could that be a problem?

Reply:
@adamchainz I thought about it, but, the less characters, the less unique. I also thought about base32 or base64 to he more efficient.  Check the Redis Keys section:
 https://redis.io/topics/data-types-intro

Reply:
@adamchainz Celery might be a good reference: https://github.com/celery/celery/blob/master/celery/app/base.py#L732

Reply:
@adamchainz I'm not sure why they use the uuid from kombu but it's pretty much UUID4: https://kombu.readthedocs.io/en/latest/reference/kombu.utils.uuid.html

Reply:
This is good but UUID's actually have some predictability to them. Using the hex value is also less efficient as you noted. For true randomness, I think it'd be best to turn to `get_random_string` from inside Django: https://github.com/django/django/blob/3c35825009540e222311d591c7330fa17dd76c99/django/utils/crypto.py#L54 . You can use length 22 to get 131 bits randomness, which is more than the 128 bits that a UUID4 uses in 32 hex characters.

Reply:
I´m only following this from the side-lines 🤹‍♂ but surely UUIDs are random enough here? 

This is just to stop naming collisions right, rather than anything _secure_ that we need to stop baddies from guessing?

Do we need to do anything more?

Reply:
@carltongibson Yeah, it's definitely not cryptographically secure, but I really don't see any reasons we should ensure cryptographically secure randomness, with access to Redis an attacker could most probably do anything, randomized keys or not. Bear in mind that cryptographically secure randomization can be considerably more expensive.

Reply:
@carltongibson To be more specific, the implementation in my PR is cryptographically secure but only for CPython: https://stackoverflow.com/a/53542384/1418283

Reply:
You're right, we don't need crypto grade security. `get_random_string` is still denser and so wins some storage space but that may not be a concern. Happy to merge as-is.